src/ansiblelint/yaml_utils.py

```diff
@@ -977,7 +992,11 @@ def dumps(self, data: Any) -> str:
                 stream.write(preamble_comment)
             self.dump(data, stream)
             text = stream.getvalue()
-        return self._post_process_yaml(text)
+        strip_version_directive = hasattr(self, "_yaml_version_default")
```

Comment:
Is this ever false? I thought ruamel.yaml had a default for `self._yaml_version_default`.

Reply:
That is there for a good reason, that attribute is not always defined!

Reply:
Doh! Of course. I added `self._yaml_version` and `self._yaml_version_default` to work around ruamel.yaml's quirky use of the `version` attribute. This PR changes that so those are only present sometimes. Hmm...